### PR TITLE
Start terraform

### PR DESCRIPTION
**Pull request by IshikaM19**
I created a terraform folder. Using terraform, in the main.tf I created a database and database security group that allows us to talk to postgres. I also created 3 ECR's...one for the pipeline, dashboard and report. I've also started working on an SQL Schema script but I still need to check that it works and the tables can be easily created and dropped.

### CODE REVIEW

File: terraform/variables.tf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    type = string
+}
+
+variable "DB_PASSWORD" {
+    type = string
+}
+
+variable "VPC_ID" {
+    type = string
+    default = "vpc-04423dbb18410aece"
```
Comment (IB64) [inline terraform/variables.tf:15]: Nice option of having a default!

File: pipeline/schema.sql [hunk 2]
```diff
@@ -0,0 +1,56 @@
+-- This file should contain table definitions for the database.
+
+DROP TABLE IF EXISTS sale_event;
+DROP TABLE IF EXISTS country;
+DROP TABLE IF EXISTS album_genre;
+DROP TABLE IF EXISTS genre;
+DROP TABLE IF EXISTS album;
+DROP TABLE IF EXISTS artist;
```
Comment (caithomas6789) [inline pipeline/schema.sql:8]: Looks like all tables are included from the ERD.

File: pipeline/schema.sql [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    artist_name VARCHAR NOT NULL UNIQUE,
+    PRIMARY KEY (artist_id)
+);
+
+CREATE TABLE album(
+    album_id INT GENERATED ALWAYS AS IDENTITY,
+    album_name VARCHAR NOT NULL UNIQUE,
+    artist_id INT NOT NULL UNIQUE,
+    PRIMARY KEY (album_id),
+    FOREIGN KEY (artist_id) REFERENCES artist(artist_id) ON DELETE CASCADE
```
Comment (caithomas6789) [inline pipeline/schema.sql:28]: I like how you included the delete cascade.

File: terraform/main.tf [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  engine_version               = "15.3"
+  instance_class               = "db.t3.micro"
+  publicly_accessible          = true
+  performance_insights_enabled = false
+  skip_final_snapshot          = true
+  db_subnet_group_name         = "public_subnet_group"
+  vpc_security_group_ids       = [aws_security_group.database-sg.id]
+  username                     = var.DB_USER
+  password                     = var.DB_PASSWORD
+}
```
Comment (caithomas6789) [inline terraform/main.tf:51]: Terraform looks correct.

**Review by caithomas6789 — APPROVED** [review]

**Pull request merged**